ee/katc/config.go

```diff
@@ -66,14 +66,15 @@ func (kst *katcSourceType) String() string {
 // JSON KATC config.
 type rowTransformStep struct {
 	name          string
-	transformFunc func(ctx context.Context, slogger *slog.Logger, row map[string][]byte) (map[string][]byte, error)
+	transformFunc func(ctx context.Context, slogger *slog.Logger, sourcePath string, row map[string][]byte) (map[string][]byte, error)
 }
 
 const (
 	snappyDecodeTransformStep       = "snappy"
 	deserializeFirefoxTransformStep = "deserialize_firefox"
 	deserializeChromeTransformStep  = "deserialize_chrome"
 	camelToSnakeTransformStep       = "camel_to_snake"
+	addUsernameFromFilePathStep     = "add_username_from_filepath"
```

Comment:
We don't have to keep the same table semantics.

Reply:
And the assumptions in `homeDirLocations` have edges cases that will break unexpectedly :|

Reply:
> edges cases that will break unexpectedly :|

In that case, the row will still be returned, just without the `username` field included! And we'd still have the `path` column available.

I think my preference is to keep this for parity with the old table, just in case we do want it. But if you feel strongly about getting rid of it, I'm happy to close this PR too!

Reply:
My hunch is we should drop it. I'm not quite sure why. Unless we cannot avoid it, I'd rather not return potentially wrong data.